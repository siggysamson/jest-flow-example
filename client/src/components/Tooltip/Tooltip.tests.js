// @flow

const React = require('react');
const { shallow } = require('enzyme');
const { shallowToJson } = require('enzyme-to-json');
const Tooltip = require('./Tooltip');

const getComponent = (props, children) => (
  <Tooltip {...props}>
    {children}
  </Tooltip>
);

const getSampleChildren = () => (
  <div>Wrapped element</div>
);

const getSampleOverlay = () => (
  <div>Sample overlay</div>
);

describe('Tooltip', () => {
  test('renders', () => {
    const tree = shallow(
      getComponent({ overlay: getSampleOverlay() }, getSampleChildren()),
    );
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test('renders with property "visible"', () => {
    const tree = shallow(getComponent({
      overlay: getSampleOverlay(),
      visible: true,
    }, getSampleChildren()));
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test('handles mouse enter', () => {
    const tree = shallow(getComponent({
      overlay: getSampleOverlay(),
    }, getSampleChildren()));

    tree.find('.tooltip__trigger').prop('onMouseEnter')();
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test('handles mouse leave', () => {
    const tree = shallow(getComponent({
      overlay: getSampleOverlay(),
      visible: true,
    }, getSampleChildren()));

    tree.find('.tooltip__trigger').prop('onMouseLeave')();
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
