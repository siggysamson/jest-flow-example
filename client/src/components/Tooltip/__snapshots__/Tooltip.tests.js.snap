exports[`Tooltip handles mouse enter 1`] = `
<div>
  <div
    className="">
    <div>
      Sample overlay
    </div>
  </div>
  <div
    className="tooltip__trigger"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    <div>
      Wrapped element
    </div>
  </div>
</div>
`;

exports[`Tooltip handles mouse leave 1`] = `
<div>
  <div
    className="tooltip--visible">
    <div>
      Sample overlay
    </div>
  </div>
  <div
    className="tooltip__trigger"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    <div>
      Wrapped element
    </div>
  </div>
</div>
`;

exports[`Tooltip renders 1`] = `
<div>
  <div
    className="">
    <div>
      Sample overlay
    </div>
  </div>
  <div
    className="tooltip__trigger"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    <div>
      Wrapped element
    </div>
  </div>
</div>
`;

exports[`Tooltip renders with property "visible" 1`] = `
<div>
  <div
    className="tooltip--visible">
    <div>
      Sample overlay
    </div>
  </div>
  <div
    className="tooltip__trigger"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    <div>
      Wrapped element
    </div>
  </div>
</div>
`;
